modules/controller/dh-controller.js

```diff
@@ -194,45 +198,122 @@ class DHController {
         }
     }
 
-    async _extendResponse(response) {
-        const missingObjects = {};
-        for (const trailElement of response) {
-            const object = trailElement.otObject;
-
-            const elementIsMissing =
-                (array, element) => !array.find(e => e.otObject['@id'] === element['@id']);
-
-            for (const relation of object.relations) {
-                if (elementIsMissing(response, relation.linkedObject)) {
-                    if (!missingObjects[relation.linkedObject['@id']]) {
-                        missingObjects[relation.linkedObject['@id']] = trailElement.datasets;
-                    } else {
-                        missingObjects[relation.linkedObject['@id']] =
-                            [...new Set(missingObjects[relation.linkedObject['@id']], trailElement.datasets)];
-                    }
-                }
-            }
+    async lookupTrail(req, res) {
+        this.logger.api('POST: Trail lookup request received.');
+
+        if (req.body === undefined ||
+            req.body.identifier_types === undefined ||
+            req.body.identifier_values === undefined
+        ) {
+            res.status(400);
+            res.send({
+                message: 'Bad request',
+            });
+            return;
         }
 
-        if (Object.keys(missingObjects).length > 0) {
-            /*
-              missingObjects: {
-                id1: [  dataset 1,  dataset 2, ... ],
-                id2: [  dataset 2,  dataset x, ... ],
-                ...
-              }
-             */
+        const { identifier_types, identifier_values, opcode } = req.body;
+
+
+        try {
+            const response = await this.trailService.lookupTrail(
+                identifier_types,
+                identifier_values,
+                opcode,
+            );
 
-            const missingIds = Object.keys(missingObjects);
-            const missingElements =
-                await this.graphStorage.findTrailExtension(missingIds, missingObjects);
+            res.status(200);
+            res.send(response);
+        } catch (e) {
+            this.logger.error(e.message);
+            res.status(501);
+            res.send({ errorMessage: 'Internal error' });
+        }
+    }
 
-            const trailExtension = this.importService.packTrailData(missingElements);
+    async findTrail(req, res) {
+        this.logger.api('POST: Trail find request received.');
 
-            return response.concat(trailExtension);
+        if (req.body === undefined ||
+            req.body.unique_identifiers === undefined
+        ) {
+            res.status(400);
+            res.send({
+                message: 'Bad request',
```

Comment:
Update this to return more descriptive message something like:
Unable to find data with given parameters! unique_identifiers is required!

Also add log so that we know why the request didn't pass

Same thing for lookupTrail